lineman/app/components/thread_page/proposal_positions_panel/proposal_positions_panel.coffee

```diff
@@ -6,7 +6,8 @@ angular.module('loomioApp').directive 'proposalPositionsPanel', ->
   controller: ($scope, Records, CurrentUser, ModalService, VoteForm) ->
 
     $scope.changeVote = ->
-      ModalService.open VoteForm, proposal: -> $scope.proposal
+      ModalService.open VoteForm, 
+        proposal: -> $scope.proposal.clone()
```

Comment:
Yep, I was just being paranoid, I think.

I've been having a look at the Modal Service and keep feeling like there's a better way to handle resolves, specifically that the form should know about what resolves it's expecting and have a sensible default (Records.proposals.build()) for when it doesn't receive one

Something like
```
# Edit Proposal:
ModalService.open ProposalForm, { proposal: $scope.proposal }

# New proposal:
ModalService.open ProposalForm, { discussion_id: $scope.discussion.id }

# ProposalForm:
$scope.proposal = options['proposal'] or Records.proposals.build(discussion_id: options['discussion_id'])
```

With some element to pass the options through in between
(It's still a half-baked idea, sorry)